packages/backend/src/tokens/authStatus.ts

```diff
@@ -27,7 +27,11 @@ export type SignedInState = {
   afterSignInUrl: string;
   afterSignUpUrl: string;
   isSignedIn: true;
-  toAuth: () => SignedInAuthObject;
+  toAuth: <TOptions extends PendingSessionOptions>(
+    opts?: TOptions,
+  ) => TOptions extends { treatPendingAsSignedOut: false }
```

Comment:
This generic makes it so that the return type of standlonone `requestState.toAuth` leads to `SignedInObject` in order to not break other modules. 

The same behavior happens in runtime, since the return type is just a union upon `treatPendingAsSignedOut: false` -> if the session exists, but has a pending status, than we override it as a signed out.